File: backend/app/schemas/review.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from .user import User
from .website import Website

class ReviewBase(BaseModel):
    content: str
    rating: int
    website_id: int

class ReviewCreate(ReviewBase):
    pass

class ReviewUpdate(BaseModel):
    content: Optional[str] = None
    rating: Optional[int] = None

class Review(ReviewBase):
    id: int
    user_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    user: Optional[User] = None
    website: Optional[Website] = None
    user_email: Optional[str] = None
    user_full_name: Optional[str] = None
    website_name: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/review.ts
// src/types/review.ts
export interface Review {
  id: number;
  content: string;
  rating: number;
  user_id: number;
  website_id: number;
  created_at: string;
  updated_at?: string;
  user_email?: string;
  user_full_name?: string;
  website_name?: string;
}

export interface ReviewCreate {
  content: string;
  rating: number;
  website_id: number;
}

export interface ReviewUpdate {
  content?: string;
  rating?: number;
}